File: app/models/MailAddress.java
package models;

import models.base.GeneratedIdentityModel;

import javax.persistence.Entity;


@Entity
public class MailAddress extends GeneratedIdentityModel {

    private String street;

    private String zip;

    private String city;

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getZip() {
        return zip;
    }

    public void setZip(String zip) {
        this.zip = zip;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    @Override
    public String toString() {
        return street + ", " + zip + "  " + city;
    }
}

File: app/models/facility/MailAddress.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.facility

import jakarta.persistence.Entity
import models.base.GeneratedIdentityModel

import scala.compiletime.uninitialized

@Entity
class MailAddress extends GeneratedIdentityModel:
  var street: String = uninitialized
  var zip: String    = uninitialized
  var city: String   = uninitialized

  override def toString: String = s"$street, $zip  $city"
